src/nbl/video/IGPUDescriptorSet.cpp

```diff
@@ -152,7 +169,10 @@ void IGPUDescriptorSet::dropDescriptors(const IGPUDescriptorSet::SDropDescriptor
 
 	// we only increment the version to detect UPDATE-AFTER-BIND and automagically invalidate descriptor sets
 	// so, only if we do the path that writes descriptors, do we want to increment version
-    if (getOriginDevice()->getEnabledFeatures().nullDescriptor)
+    auto& bindingRedirect = m_layout->getDescriptorRedirect(descriptorType);
+    auto bindingCreateFlags = bindingRedirect.getCreateFlags(bindingRedirect.findBindingStorageIndex({ drop.binding }));
+    bool shouldIncrementVersion = not (bindingCreateFlags.hasFlags(IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_UPDATE_AFTER_BIND_BIT) or bindingCreateFlags.hasFlags(IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_UPDATE_UNUSED_WHILE_PENDING_BIT));
+    if (getOriginDevice()->getEnabledFeatures().nullDescriptor or shouldIncrementVersion)
```

Comment:
AND, not OR

alternatively for consistent behaviour, you can just do `shouldIncrementVersion` and not care about `nullDescriptor` feature